extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -265,6 +269,28 @@ public String deleteExtensionMetadata(final String extensionName) throws Validat
         }
     }
 
+    public String registerExtensionMetadata(final String extensionName, final String path, final String description)
+        throws URISyntaxException, FalconException, IOException{
+        Configuration conf = new Configuration();
+        URI uri = new URI(path);
+        conf.set("fs.default.name", uri.getScheme() + "://" + uri.getAuthority());
+        FileSystem fileSystem =  HadoopClientFactory.get().createFalconFileSystem(uri);
+        FileStatus[] fileStatusReadMe = fileSystem.listStatus(new Path(uri.getPath() + "/README"));
+        PathFilter filter=new PathFilter(){
+            public boolean accept(Path file){
+                return file.getName().endsWith(".jar");
+            }
+        };
+        FileStatus[] fileStatusJar = fileSystem.listStatus(new Path(uri.getPath() + "/libs"), filter);
+
+        if (fileStatusReadMe.length == 1 && fileStatusJar.length >0){
+            metaStore.storeExtensionMetadataBean(extensionName, path, ExtensionType.CUSTOM, description);
+        }else{
+            throw new ValidationException("Packaging structure is not currect please refer to documentation");
```

Comment:
Error message : Packaging structure of extension ${extensionName} is not correct. Please refer the documentation for details